src/libflake-c/nix_api_flake.h

```diff
@@ -238,6 +238,18 @@ void nix_flake_reference_free(nix_flake_reference * store);
 nix_value * nix_locked_flake_get_output_attrs(
     nix_c_context * context, nix_flake_settings * settings, EvalState * evalState, nix_locked_flake * lockedFlake);
 
+/**
+ * @brief Get the root path of a flake.
```

Comment:
Note that when lazy trees are enabled,  `flake.path` is a virtual (i.e. random) path that doesn't actually exist. So this is probably not a useful thing to expose.

Depending on the intended use, you could:
* Copy the path to the Nix store by calling `state.devirtualize()`. This can be expensive.
* Create C bindings for the `SourceAccessor` / `SourcePath` API to allow the caller to access the flake lazily.

Reply:
Hmm, that's a good point. Is there a way to detect when lazy trees is active? I know I somewhat need a real file since I'm calling `std::fs::read_to_string()` within rust.